src/main/java/org/datadog/jenkins/plugins/datadog/configuration/DatadogAgentConfiguration.java

```diff
@@ -77,24 +79,28 @@ public DatadogClient createClient() {
 
     @Override
     public void validateTracesConnection() throws Descriptor.FormException {
+        FormValidation connectivityCheckResult = checkTracesConnectivity(agentHost, agentTraceCollectionPort);
+        if (connectivityCheckResult.kind == FormValidation.Kind.ERROR) {
+            throw new Descriptor.FormException("CI Visibility connectivity check failed: " + connectivityCheckResult.getMessage(), "ciVisibilityData");
+        }
+    }
+
+    private static FormValidation checkTracesConnectivity(String agentHost, Integer agentTraceCollectionPort) {
         if (StringUtils.isBlank(agentHost)) {
-            throw new Descriptor.FormException("CI Visibility requires agent host to be set", "agentHost");
+            return FormValidation.error("Host name missing");
         }
         if (agentTraceCollectionPort == null) {
-            throw new Descriptor.FormException("CI Visibility requires agent trace collection port to be set", "agentTraceCollectionPort");
-        }
-        String errorMessage = checkConnectivity(agentHost, agentTraceCollectionPort);
-        if (errorMessage != null) {
-            throw new Descriptor.FormException("CI Visibility connectivity check failed: " + errorMessage, "ciVisibilityData");
+            return FormValidation.error("Trace collection port missing");
         }
-    }
-
-    private static String checkConnectivity(final String host, final int port) {
-        try (Socket ignored = new Socket(host, port)) {
-            return null;
-        } catch (Exception ex) {
-            DatadogUtilities.severe(logger, ex, "Failed to create socket to host: " + host + ", port: " + port);
-            return ex.getMessage();
+        try {
+            Set<String> endpoints = DatadogAgentClient.fetchAgentEndpoints(new HttpClient(10_000), agentHost, agentTraceCollectionPort);
```

Comment:
Can we have a constant to name that `10_000`?

Also, I'm seeing that this is the `timeout`. This means that if the plugin cannot communicate with the Datadog Agent, the request can take 10 seconds? This might be confusing for customers if there's no indication that the plugin is doing something under the hood.

If that's the case, could we limit the timeout to something lower (1 second? 2 seconds at most?) to provide quicker feedback.